handshake_ix.go

```diff
@@ -99,11 +99,11 @@ func ixHandshakeStage0(f *Interface, hh *HandshakeHostInfo) bool {
 	return true
 }
 
-func ixHandshakeStage1(f *Interface, addr netip.AddrPort, via *ViaSender, packet []byte, h *header.H) {
```

Comment:
this is also a nice step towards using different types to represent overlay vs underlay IPs, which I think could make nebula much easier to read :D